pkg/controller/controller_test.go

```diff
@@ -457,3 +457,81 @@ func TestGetEmail(t *testing.T) {
 		})
 	}
 }
+
+func TestGetCatagoriesDisplayData(t *testing.T) {
+	t.Parallel()
+
+	validators := map[string]jvspb.Validator{
+		"jira": &mockValidator{
+			Valid:       true,
+			DisplayName: "Jira issue key",
+			Hint:        "Jira Issue key under JVS project",
+		},
+		"git": &mockValidator{
+			Valid:       false,
+			DisplayName: "Git issue key",
+			Hint:        "Git Issue key under JVS project",
+		},
+	}
+	jiraUIData := &jvspb.UIData{
+		DisplayName: "Jira issue key",
+		Hint:        "Jira Issue key under JVS project",
+	}
+	gitUIData := &jvspb.UIData{
+		DisplayName: "Git issue key",
+		Hint:        "Git Issue key under JVS project",
+	}
+	expectedCatagoriesData := map[string]*jvspb.UIData{
+		"jira": jiraUIData,
+		"git":  gitUIData,
+	}
+
+	cases := []struct {
+		name       string
+		validators map[string]jvspb.Validator
+		wantRes    map[string]*jvspb.UIData
+		wantErr    string
+	}{
+		{
+			name:       "empty_validators",
+			validators: make(map[string]jvspb.Validator),
+			wantRes:    make(map[string]*jvspb.UIData),
+		},
+		{
+			name:       "success_when_validator_with_ui_data",
+			validators: validators,
+			wantRes:    expectedCatagoriesData,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			gotRes, err := getCatagoriesDisplayData(tc.validators, context.Background())
+			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+				t.Errorf("Unexpected err: %s", diff)
+			}
+
+			// Compares goRes and wantRes
+			wantResLen := len(tc.wantRes)
+			gotResLen := len(gotRes)
+			if gotResLen != wantResLen {
+				t.Errorf("Expected map with size '%d' but actual map with size '%v'", wantResLen, gotResLen)
+			}
+
+			for k, w := range tc.wantRes {
+				g, ok := gotRes[k]
+				if !ok {
+					t.Errorf("Key '%s' not found in the actual map", k)
+				}
+
+				if w.DisplayName != g.DisplayName || w.Hint != g.Hint {
+					t.Errorf("Values for key '%s' are different. Expected: %v, Actual: %v", k, w, g)
+				}
```

Comment:
Use `cmp`.

```go
if diff := cmp.Diff(want, got, protocmp.Transform()); diff != "" {
  t.Errorf("Display data (-want,+got):\n%s", diff)
}
```

Reply:
Thank you for the suggestion. I updated this part in the latest version.